fvtest/porttest/omrsockTest.cpp

```diff
@@ -1285,26 +1288,33 @@ TEST(PortSockTest, poll_functionality_many_sockets)
 	}
 	ASSERT_EQ(rc, 1);
 
-	for (int32_t i = 0; i < 10; i++) {
-		OMRPORTLIB->sock_get_pollfd_info(OMRPORTLIB, &pollArray[i], &socketPoll, &revents);
-		if (0 != (revents & OMRSOCK_POLLOUT)) {
-			ASSERT_EQ(clientSocket, socketPoll);
-			bytesSent = OMRPORTLIB->sock_send(OMRPORTLIB, clientSocket, (uint8_t *)msg, strlen(msg) + 1, 0);
-			ASSERT_GT(bytesSent, 0);
-			break;
-		}
-	}
+	/* The client socket should be ready to write */
+	OMRPORTLIB->sock_get_pollfd_info(OMRPORTLIB, &pollArray[CLIENT_SOCKET_POLL_IDX], &socketPoll, &revents);
+	ASSERT_EQ(clientSocket, socketPoll);
+	ASSERT_NE(revents & OMRSOCK_POLLOUT, 0);
+
+	bytesSent = OMRPORTLIB->sock_send(OMRPORTLIB, clientSocket, (uint8_t *)msg, strlen(msg) + 1, 0);
+	ASSERT_GT(bytesSent, 0);
+
+
+	/* Ignore OMRSOCK_POLLOUT on client socket, since its state is indeterminate/ */
+	/* Listen for HUP, since an error is returned if sock_pollfd_init recieves an event arg of 0. */
+	 EXPECT_EQ(OMRPORTLIB->sock_pollfd_init(OMRPORTLIB, &pollArray[CLIENT_SOCKET_POLL_IDX], clientSocket, OMRSOCK_POLLHUP), 0);
 
 	/* Check that server POLLIN is ready. Give poll up to 10 times to be ready. */
+	bool succeeded = false;
 	for (int32_t i = 0; i < 10; i++) {
-		if (2 == (rc = OMRPORTLIB->sock_poll(OMRPORTLIB, pollArray, 10, 1000))) {
```

Comment:
`sock_poll` returns the number of sockets ready for IO. in the new changes, `(rc == 2)` is changed to `(rc > 0)`. why was this change required? should the test fail for `(rc == 1)`?

Reply:
The client socket isn't guaranteed to be ready to write again. In addition, its possible that the OS is smart enough to skip the network stack and the socket buffers are shared (eg client write buffer == server read buffer). If that is the case, and the buffer is full, the client socket will not be ready to write again until the server reads to free up space in the buffer.